### DISCUSSION TITLE
Run error: ModuleNotFoundError: No module named 'pandas'

### DISCUSSION THREAD

**Discussion by wanyingw193** [NREL/ReEDS-2.0#50] [Q&A: Execution]
To whom it may concern, 

When I try to run ReEDS, it always throws the following error. But if I run WriteHintage.py alone, it does not produce any errors. What other files can I check to debug this error? Thanks!

Error from gamslog.txt:
File "C:\software\ReEDS_OpenAccess-SM_EasySetup\\input_processing\WriteHintage.py", line 14, in <module>
    import pandas as pd
ModuleNotFoundError: No module named 'pandas'

**Comment by vcelloho** [comment]
Try running from command prompt.
pip install pandas

If this doesn't work let me know.

**Reply by wanyingw193** [reply]
Hi,

I run that, and add the installed location to both system environment and .brashrc, but it did not work...

**Reply by maxxb77** [reply]
Hi there -

Sounds like there are two different versions of python installed on your machine - a nice trick is to create a file 'install.py' with just the two lines:
_import pip
pip.main(['install', 'pandas'])_

-then- create a file install.gms with just the single line:
_$call 'python install.py'_

-then- from the command prompt, call:
_gams install.gms_

It's a bit of a workaround but it installs the package for whichever version of python your call to GAMS is seeing. For some reason I've seen this more frequently on unix/linux-based systems and figured it might be the case given your mention of .bashrc in the previous message

**Reply by wanyingw193** [reply]
Hi,

Sorry that questions keep coming up.

Above solution works for calling pandas (many thanks!). But the program only produces [testname]_2010.lst, and reeds_data_[testname]_2012.gms, then it stops. No error file is shown in ReEDS_Augur, and no error in .lst. This is a bit confusing and I wonder how to find the errors?

**Comment by vcelloho** [comment]
There's three places to look, the first is to be certain that there isn't an error you overlooked in the [testname]2010.lst file. I usually search for this by searching for **** in the document. The second thing to search the gamslog.txt, this can be helpful in identifying error messages that might have flashed by too quickly that didn't appear in the .lst file or as part of augur. Finally in your cases file verify that you haven't specified an early endyear.

**Reply by wanyingw193** [reply]
Hi, I just wanted to update that ReEDS started working after I deleted the endyear row in the case_test.csv, previously I had set 2035 as the endyear. And there are no errors in the gamslog.txt and .lst files.

Thank you so much!

**Comment by zhxn30663** [comment]
Hi @wanyingw193,
I just met the same issue. This seems to caused by the incorrect environment variable setting. Maybe ‘%GAMSDIR%' in the 'PATH' has a higher priority. This makes anaconda envs to be called correctly in CMD or Powershell but GAMS does not run properly. You may change the priority of %GAMSDIR%.  Or open CMD (do not use Powershell), and cd to \WHERE_GAMS_INSTALLED\GMSPython\Scripts. Then run 'pip install pandas' for the embedded python in GAMS.
